tests/test_language_metadata.py

```
@@ -0,0 +1,922 @@
+    fake_module.DetectorFactory = FakeDetectorFactory
+    fake_module.LangDetectException = FakeLangDetectException
+    fake_module.detect_langs = fail_detect
+    monkeypatch.setitem(sys.modules, "langdetect", fake_module)
+
+    assert LangdetectLanguageDetector().detect("???").language == "unknown"
+
+
+def test_langdetect_provider_reports_missing_dependency(monkeypatch):
+    original_import = __import__
+
+    def fake_import(name, *args, **kwargs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `fake_import`

(ANN202)